pkg/registry/batch/job/strategy.go

```diff
@@ -147,6 +136,13 @@ func (jobStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object
 	if !apiequality.Semantic.DeepEqual(newJob.Spec, oldJob.Spec) {
 		newJob.Generation = oldJob.Generation + 1
 	}
+
+	// While legacy tracking is supported, we use an annotation to mark whether
```

Comment:
this prevents dropping but doesn't prevent adding on update like we previously did... commented above about doing both

Reply:
Adding would fail on validation, like it always did.